### DISCUSSION TITLE
How to define User Group Permission Rule with multiple permissions

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#217] [Q&A] [answered]
Hi!
I'm trying to create a user group permission rule on multiple permissions like this documented example:
`["write:orders", "view:orders"] in permissions;`
(source: https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/endpoints#manage-the-security-of-your-endpoints)

The `authorization-service` is returning an error:
```
    expression: "[\"permA\", \"permB\"] in permissions"
    methods: "GET"
    path: "/my-path"
    error: "error creating expression program to evaluate: type-check failed due to: ERROR: <input>:1:28: found no matching overload for '@in' applied to '(list(string), map(string, bool))'\n | [\"permA\", \"permB\"] in permissions\n | ...........................^"
```

What is the correct way to define this type of rule?

Thanks

**Comment by davidebianchi** [comment, marked as answer]
Hi @elmalakomar, I think that the documentation is not correct. Permissions added are handled as a map, so the check should be done:
```
"permA" in permissions && "permB" in permissions
```

or

```
permissions["permA"] && permissions["permB"]
```

Do you want to support the `in` operator also on multiple permissions? I don't know if it's possible to add it.

**Reply by elmalakomar** [reply]
Hi @davidebianchi ,
thanks for the response.
If it's possible, having this support could be helpful and more clear when there are a lot of permissions to handle, as described in the doc referenced above (permissions in OR).
Until that, I will work with your solution
